src/core/components/hv-route/index.tsx

```diff
@@ -98,6 +98,29 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, ScreenState> {
       this.load();
       this.needsLoad = false;
     }
+
+    // Clean out loading screen cache after the screen has been rendered
+    if (
+      this.props.element ||
+      this.localDoc ||
+      this.props.route?.params?.isModal
```

Comment:
We want to clean up the cache _after_ the loading UI was removed.

A couple issues with the implementation:
- This will render every time any props changes including other props that are not part of this condition. This condition might be true multiple times, meaning this code will run more than what we need it to. Typically, we would need to compare the prevProps to only run the code when we know the condition to render the LoadScreen has changed.
- The condition needs to match exactly what's in the `render` method for this code to continue working in the future - it would be quite easy to break it if we don't pay attention, so ideally the condition should be shared between this code and the code in `render`. However I suspect this will become quite complicated to manage given that we also need to account for `prevProps`.

Looking at the implementation for the render method, we render a local `Load` component instance that is created either from the `loadingScreen` prop or the internal `Loading` component. Any way to wrap the creation of this component with another component that will implement in its "unmount" method the clean up for the cache?

Reply:
That's a good suggestion, I'll try that out and see if I can encapsulate the logic there